cla_public/templates/macros/form.html

```diff
@@ -1,116 +1,217 @@
+    {%- if field %}id="field-{{ field.id }}"{% endif -%}
+  >
+    {% if legend_ %}
+      <legend class="form-group-label" {% if field %} id="field-label-{{ field.id }}"{% endif %}>
+        <span>{{ legend_ }}</span>
+      </legend>
+    {% endif %}
+
+    {{ render_field_errors(field.errors) if field and field.errors}}
+    {{ render_field_description(field) }}
+
+    {% if field or caller %}
+      <div class="form-row {{ kwargs.row_class }}">
+        {% if field and not kwargs.field_as_label %}
+          {{ render_field(field, field_attrs) }}
+        {% endif %}
+        {% if caller %}
+          {{ caller() }}
+        {% endif %}
+      </div>
+    {% endif %}
+  </fieldset>
+{% endmacro %}
+
+
+{#
+  Renders `form-group` wrapper for form fields
+
+  Params:
+    - field <object> (optional)
         WTForm field
-    - is_subfield <boolean> (default: False)
-        Whether to treat the fieldset as subfield
-    - is_hidden <boolean> (default: False)
-        Whether to hide the field (CSS)
-    - controlled_by <object> (default: None)
-        WTForm field which controls the visibility of the fieldset
-    - control_value <string> (default: '1')
-        Input value which shows the fieldset when toggled
-    - as_label <boolean> (default: False)
-        Whether fieldset legend should contain label element
-    - class_ <string> (optional)
-        Fieldset CSS class
-    - has_label <boolean> (default: True)
-        Whether to show the `legend/label` elements
-    - use_row <boolean> (default: True)
-        Whether to wrap the 'caller' in `.form-row` div
-    - persist_values <boolean> (default: False)
-        Whether to persist input values when fieldset is hidden
+    - class_ <string> (default: '')
+        Additional CSS class (all form group wrappers get 'form-group' CSS class)
+    - field_attrs <object> (default: {})
+        Custom HTML attributes for field
+    - controlled_by <string> (default: None)
+        name of the field that controls the visibility of this container
+    - control_value <string> (default: None)
+        value of the field that controls the visibility of this field
+    - row_class <string> (default: '')
+        Optional CSS class for `form-row` element
 #}
-{% macro fieldset(
-  field=None,
-  is_subfield=False,
-  is_hidden=False,
-  controlled_by=None,
-  control_value='1',
-  as_label=False,
-  class_='',
-  has_label=True,
-  use_row=True,
-  persist_values=False,
-  role=None
-) %}
-  {% if field and field.description %}
-    {% set describedby = field.id %}
-  {% else %}
-    {% set describedby = kwargs.describedby %}
-  {% endif %}
-  <fieldset class="{% if class_ %}{{ class_ }}{% else %}form-group{% endif %}
-    {%- if field and field.errors %} m-error{% endif %}
-    {%- if is_subfield %} form-subfield{% endif %}
-    {%- if is_hidden == True or controlled_by and control_value %} s-hidden{% endif %}"
+{% macro group(field=None, class_='', field_attrs={}) %}
```

Comment:
I guess I was trying to make the macro look neater by reducing the big list of arguments it takes.